ipaserver/install/installutils.py

```diff
@@ -380,33 +380,55 @@ def update_file(filename, orig, subst):
         return 1
 
 
-def set_directive(filename, directive, value, quotes=True, separator=' ',
-                  quote_char='\"'):
+def quote_directive_value(value, quote_char):
+    """Quote a directive value
+    :param value: string to quote
+    :param quote_char: character which is used for quoting. All prior
+        occurences will be escaped before quoting to avoid unparseable value.
+    :returns: processed value
+    """
+    if value.startswith(quote_char) and value.endswith(quote_char):
+        return value
+
+    return "{quote}{value}{quote}".format(
+        quote=quote_char,
+        value="".join(ipautil.escape_seq(quote_char, value))
+    )
+
+
+def unquote_directive_value(value, quote_char):
+    """Unquote a directive value
+    :param value: string to unquote
+    :param quote_char: character to strip. All escaped occurences of
+        `quote_char` will be uncescaped during processing
+    :returns: processed value
+    """
+    unescaped_value = "".join(ipautil.unescape_seq(quote_char, value))
+    return unescaped_value.strip(quote_char)
```

Comment:
Using `strip()` causes issues when the actual value starts or ends with the same type of quote as is used to escape it, e.g. `'cn=\'ABC\''` gets stripped to `cn='ABC`.